gix-ref/src/store/packed/buffer.rs

```diff
@@ -84,6 +68,35 @@ pub mod open {
                 path,
             })
         }
+
+        /// Open the file at `path` and map it into memory if the file size is larger than `use_memory_map_if_larger_than_bytes`.
+        ///
+        /// In order to allow fast lookups and optimizations, the contents of the packed refs must be sorted.
+        /// If that's not the case, they will be sorted on the fly with the data being written into a memory buffer.
+        pub fn open(path: PathBuf, use_memory_map_if_larger_than_bytes: u64) -> Result<Self, Error> {
+            let backing = if std::fs::metadata(&path)?.len() <= use_memory_map_if_larger_than_bytes {
+                packed::Backing::InMemory(std::fs::read(&path)?)
+            } else {
+                packed::Backing::Mapped(
+                    // SAFETY: we have to take the risk of somebody changing the file underneath. Git never writes into the same file.
+                    #[allow(unsafe_code)]
+                    unsafe {
+                        Mmap::map(&std::fs::File::open(&path)?)?
+                    },
+                )
+            };
+            Self::open_with_backing(backing, path)
+        }
+
+        /// Open a buffer from existing memory.
+        ///
+        /// In order to allow fast lookups and optimizations, the contents of the packed refs must be sorted.
+        /// If that's not the case, they will be sorted on the fly with the data being written into a memory buffer.
+        #[cfg(fuzzing)]
+        pub fn open_from_memory(bytes: &[u8]) -> Result<Self, Error> {
```

Comment:
As this adds a feature, could you put this change into a separate commit and prefix the subject line with `feat:` ? The subject itself could say which method was added, with the body being a short description of what it does. 

That automatically goes into the changelog.

Reply:
Done